### PR TITLE
Hotfix/alert in departure list

### PR DESCRIPTION
**Pull request by pnuts2**

### CODE REVIEW

File: Classes/model/CoreData/StopLocation.h [hunk 1]
```diff
@@ -68,6 +69,13 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (NSArray<Alert *> *)alertsIncludingChildren;
 
+/**
+ This method is used to reset any alerts that are currently associated
+ with the StopLocation. This is usually called after alerts are altered
+ in core data, e.g., as a result of parsing alerts in a server response.
+ */
+- (void)clearAlerts;
```
Comment (nighthawk) [inline Classes/model/CoreData/StopLocation.h]: This is a confusing name. I'd expect this to remove alerts, i.e., clear the `alertHashCodes` array. Maybe something like `resetAlertsCache` is better.

File: Classes/model/CoreData/StopLocation.m [hunk 2]
```diff
@@ -211,11 +221,15 @@ - (void)clearVisits
 - (NSArray *)alertsIncludingChildren
 {
   if (!_alertsIncludingChildren) {
-    NSMutableArray *alerts = [NSMutableArray array];
-    [alerts addObjectsFromArray:[Alert fetchAlertsForStopLocation:self]];
+    NSMutableArray *alerts = [NSMutableArray array];    
+    if (self.alertHashCodes.count != 0) {
```
Comment (nighthawk) [inline Classes/model/CoreData/StopLocation.m]: Doesn't this check make more sense in `fetchAlertsForStopLocation`? Just return an empty array if there are no hash codes.

File: Classes/Loc+TripKit.swift [hunk 3]
```diff
@@ -74,6 +78,18 @@ extension Loc {
...
     }
   }
+  
+  public static func Alerts(_ count: Int) -> String {
+    switch count {
+    case 0: return NSLocalizedString("No alert", tableName: "TripKit", bundle: .tripKit, comment: "Number of alerts, in this case, there is none")
+      
+    case 1: return NSLocalizedString("1 alert", tableName: "TripKit", bundle: .tripKit, comment: "Number of alerts, in this case, there is only 1 (singular)")
+      
+    default:
```
Comment (nighthawk) [inline Classes/Loc+TripKit.swift]: Do we really ever need the cases for 0 and 1? Could just return an optional string.

**Comment by pnuts2** [comment]
previous comments all addressed + additional commits, mainly moved alert info view to TripKit

File: TripKitUI/views/TKAlertInfoView.swift [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  /// Indicating whether the alert info view is currently showing the full
+  /// content or only the title.
+  private(set) var isShowingFullContent = true
+  
+  
+  /// The alert used to configure the alert info view. Specifically, the
+  /// `title` and `text` properties will be used by the titleLabel and
+  /// instructionLabel respectively.
+  ///
+  public var alert: Alert? {
```
Comment (nighthawk) [inline TripKitUI/views/TKAlertInfoView.swift]: Please use a protocol here and then where TKAlertInfoView is used, conform the corresponding model class to the protocol. Main reason for that is that we have non-CoreData alerts, too.

**Comment by pnuts2** [comment]
updated as instructed.

**Comment by nighthawk** [comment]
One more things: Since you're adding files, please also update the Xcode project in the Project folder. That way CI will make sure it compiles and those classes will then also be made available for developers using TripKit via Carthage.

**Pull request merged**